packages/graphql/src/services/sessions.ts

```diff
@@ -879,171 +910,176 @@ export async function deactivateSessionBlock(
+                create: {
+                  type: 'SESSION',
+                  participant: {
+                    connect: { id },
+                  },
+                  score: parseInt(score),
+                  sessionParticipation: {
+                    connectOrCreate: {
+                      where: {
+                        courseId_participantId: {
+                          courseId: session.courseId as string,
+                          participantId: id,
                         },
                       },
-                      participant: {
-                        connect: {
-                          id,
+                      create: {
+                        course: {
+                          connect: {
+                            id: session.courseId!,
+                          },
+                        },
+                        participant: {
+                          connect: {
+                            id,
+                          },
                         },
                       },
                     },
                   },
                 },
-              },
-              update: {
-                score: parseInt(score),
-              },
-            })),
-          }
-        : undefined,
-    },
-    include: {
-      blocks: {
-        orderBy: {
-          order: 'asc',
+                update: {
+                  score: parseInt(score),
+                },
+              })),
+            }
+          : undefined,
+      },
+      include: {
+        blocks: {
+          orderBy: {
+            order: 'asc',
+          },
         },
       },
-    },
-  })
+    })
 
-  ctx.pubSub.publish('runningSessionUpdated', {
-    sessionId,
-    block: null,
-  })
+    ctx.pubSub.publish('runningSessionUpdated', {
+      sessionId,
+      block: null,
+    })
 
-  ctx.emitter.emit('invalidate', {
-    typename: 'Session',
-    id: session.id,
-  })
+    ctx.emitter.emit('invalidate', {
+      typename: 'Session',
+      id: session.id,
+    })
 
-  // const leaderboardUpdates = Object.entries(sessionLeaderboard).map(
-  //   ([id, score]) => {
-  //     out
-  //   }
-  // )
+    if (!isScheduled && scheduledJobs[sessionBlockId]) {
+      await scheduledJobs[sessionBlockId].cancel()
+      delete scheduledJobs[sessionBlockId]
+    }
 
-  if (!isScheduled && scheduledJobs[sessionBlockId]) {
-    await scheduledJobs[sessionBlockId].cancel()
-    delete scheduledJobs[sessionBlockId]
-  }
+    unlinkCachedBlockResults({
+      ctx,
+      sessionId,
+      sessionBlockId,
+      activeInstanceIds,
+    })
 
-  unlinkCachedBlockResults({
-    ctx,
-    sessionId,
-    sessionBlockId,
-    activeInstanceIds,
-  })
+    return updatedSession
+  } catch (error) {
+    await sendTeamsNotifications(
+      'graphql/deactivateSessionBlock',
+      `ERROR - failed to deactivate session block ${sessionBlockId} in session ${session.id} with active block ${session.activeBlockId}: ${error}`
```

Comment:
@sjschlapbach tested that this outputs something, i.e., error is not an object that cannot be printed? or should we use error.message or similar?

Reply:
Using error.message would probably be a good idea, will be fixed